x-pack/plugins/reporting/server/export_types/printable_pdf_v2/lib/generate_pdf.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { groupBy } from 'lodash';
+import * as Rx from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+import { LocatorParams } from '../../../../common/types';
+import { getRedirectAppPathHome } from '../../../../common/constants';
+import { ReportingCore } from '../../../';
+import { LevelLogger } from '../../../lib';
+import { createLayout, LayoutParams } from '../../../lib/layouts';
+import { ScreenshotResults } from '../../../lib/screenshots';
+import { ConditionalHeaders } from '../../common';
+import { getFullUrls } from '../../common/v2/get_full_urls';
+import { PdfMaker } from './pdf';
+import { getTracker } from './tracker';
+
+const getTimeRange = (urlScreenshots: ScreenshotResults[]) => {
+  const grouped = groupBy(urlScreenshots.map((u) => u.timeRange));
+  const values = Object.values(grouped);
+  if (values.length === 1) {
+    return values[0][0];
+  }
+
+  return null;
+};
+
+export async function generatePdfObservableFactory(reporting: ReportingCore) {
```

Comment:
Same feedback on this file as with the `png_v2/generate_png` file - this could be shared between the v1 and v2 PDF export types.